File: backend/controllers/auth_controller.py
from flask import Blueprint, request, jsonify
from models import UserModel, CustomerModel

auth_bp = Blueprint('auth', __name__)

@auth_bp.route('/api/login', methods=['POST'])
def admin_login():
    """Handles admin login."""
    data = request.json
    user = UserModel.authenticate(data['username'], data['password'])
    if user:
        return jsonify({'success': True, 'message': 'Admin login successful'})
    return jsonify({'success': False, 'message': 'Invalid admin credentials'}), 401

@auth_bp.route('/api/customer/signup', methods=['POST'])
def customer_signup():
    data = request.json
    customer = CustomerModel.add_customer(data['username'], data['password'], data['name'])
    if customer:
        return jsonify({'success': True, 'message': 'Signup successful', 'user': customer}), 201
    return jsonify({'success': False, 'message': 'Username already exists'}), 400

@auth_bp.route('/api/customer/login', methods=['POST'])
def customer_login():
    data = request.json
    customer = CustomerModel.authenticate(data['username'], data['password'])
    if customer:
        return jsonify({'success': True, 'message': 'Login successful', 'user': customer})
    return jsonify({'success': False, 'message': 'Invalid username or password'}), 401

File: backend/src/controllers/authController.js
const authService = require("../services/authService");

async function adminLogin(req, res) {
  const user = await authService.authenticateAdmin(req.body.username, req.body.password);
  if (!user) {
    res.status(401).json({ success: false, message: "Invalid admin credentials" });
    return;
  }

  res.json({ success: true, message: "Admin login successful" });
}

async function customerSignup(req, res) {
  const customer = await authService.signupCustomer(req.body);
  if (!customer) {
    res.status(400).json({ success: false, message: "Username already exists" });
    return;
  }

  res.status(201).json({ success: true, message: "Signup successful", user: customer });
}

async function customerLogin(req, res) {
  const customer = await authService.authenticateCustomer(req.body.username, req.body.password);
  if (!customer) {
    res.status(401).json({ success: false, message: "Invalid username or password" });
    return;
  }

  res.json({ success: true, message: "Login successful", user: customer });
}

module.exports = {
  adminLogin,
  customerLogin,
  customerSignup
};

